dotCMS/src/main/java/com/dotcms/saml/Attributes.java

```diff
@@ -0,0 +1,173 @@
+package com.dotcms.saml;
+
+import java.io.Serializable;
+
+/**
+ * Encapsulates the attributes retrieve from the Saml Assertion
+ *
+ * @author jsanca
+ */
+public class Attributes implements Serializable {
+
+	private static final long serialVersionUID = 1836313856887837731L;
+
+	// user email from opensaml
+	private final String email;
+
+	// user last name from opensaml
+	private final String lastName;
+
+	// user first name from opensaml
+	private final String firstName;
+
+	// true if opensaml returned roles
+	private final boolean addRoles;
+
+	// Saml object with the roles info.
+	private final Object roles;
+
+	// Saml object with the NameID.
+	private final Object nameID;
+
+	// SAML Session Index
+	private final String sessionIndex;
+
+	private Attributes(final Builder builder) {
+
+		this.email        = builder.email;
+		this.lastName     = builder.lastName;
+		this.firstName    = builder.firstName;
+		this.addRoles     = builder.addRoles;
+		this.roles        = builder.roles;
+		this.nameID       = builder.nameID;
+		this.sessionIndex = builder.sessionIndex;
+	}
+
+	public String getEmail()
+	{
+		return email;
+	}
+
+	public String getLastName()
+	{
+		return lastName;
+	}
+
+	public String getFirstName()
+	{
+		return firstName;
+	}
+
+	public boolean isAddRoles()
+	{
+		return addRoles;
+	}
+
+	public Object getRoles()
+	{
+		return roles;
+	}
+
+	public Object getNameID()
+	{
+		return nameID;
+	}
+
+	public String getSessionIndex() {
+		return sessionIndex;
+	}
+
+	@Override
+	public String toString() {
+		return "AttributesBean{" + "nameID='" + nameID + '\'' + ", email='" + email + '\'' + ", lastName='" + lastName + '\''
+				+ ", firstName='" + firstName + '\'' + ", addRoles=" + addRoles + ", roles=" + roles + '}';
+	}
+
+	public static final class Builder {
+		String email     = "";
+		String lastName  = "";
+		String firstName = "";
+		boolean addRoles = false;
+		Object roles     = null;
+		Object nameID    = null;
```

Comment:
Codacy found an issue: [Avoid using redundant field initializer for 'nameID'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5956529)